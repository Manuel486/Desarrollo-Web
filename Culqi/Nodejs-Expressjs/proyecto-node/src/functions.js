export const getTitle = () => "Curso de Node JS por Sergie Code";

export const getAuthor = (author) => `El autor del curso es ${author}`;

// module.exports = {
//   getTitle,
//   getAuthor,
// };
