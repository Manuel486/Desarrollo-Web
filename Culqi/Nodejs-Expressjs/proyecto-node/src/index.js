// Antigua forma - Legacy
// const { getTitle, getAuthor } = require("./functions");
// const { curso } = require("./objects");

// Nueva forma desde ES6
// Para que funciona debo ir al package.json y poner el type:module
// Tambien modificar la forma de exportar
// import { getTitle, getAuthor } from "./functions.js";
// import { curso } from "./objects.js";

// const { nombre, autor } = curso;

// console.log(getTitle());
// console.log(getAuthor("Sergio Code"));

// console.log(process); // Informa del proceso de node

// -----------------------

import { config } from "dotenv";
import env from "env-var";

config(); // Hace que se ejecute una funci√≥n que agarra las variables de entorno
const PORT = env.get("PORT").required().asPortNumber();

console.log(PORT);

console.log(process.env.PORT);
