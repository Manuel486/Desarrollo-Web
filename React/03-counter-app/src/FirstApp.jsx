// import { Fragment } from "react"
// es lo mismo que usar <></>
// es un agrupar de otros html

// No podemos mandar el objeto
const newMessage = {
  message: "Hola mundo",
  title: "Manuel",
};

// las funciones pueden ir dentro del FirstApp
// const getResult = (a,b) => {
//   return  a+b;
// }

import PropTypes from "prop-types";

export const FirstApp = ({ title, subTitle, name }) => {
  return (
    // Uso de fragmentos
    <>
      {/* <h1>{ getResult(1,2) }</h1> */}
      <h1 data-testid="test-title">{title}</h1>

      {/* Hacemos esto porque no se puede enviar un objeto a un componente */}
      {/* <code>{ JSON.stringify(newMessage) }</code>  */}

      <p>{subTitle}</p>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  );
};

// Si trabajo en vite necesito instalar el propTypes
// npm install prop-types

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
  title: "No hay titulo",
  subTitle: "",
  name: "Manuel Chunga",
};
