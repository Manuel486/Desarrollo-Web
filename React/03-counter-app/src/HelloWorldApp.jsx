// export function App() {
//   // document.createElement....
//   return <h1>Hola Mundo</h1>;
// }

// Functional component
export const HelloWorldApp = () => {
  return <h1>HelloWorld App !!</h1>;
};

// TambiÃ©n puedo exportar por default

// rafc -> Snippet para crear un componente
