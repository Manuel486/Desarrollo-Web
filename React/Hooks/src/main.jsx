import React from "react";
import ReactDOM from "react-dom/client";
import AppTask from "./components/Task/AppTask";
import EjemploComponente from "./components/UseEffect/EjemploComponente";
import { StoreApi } from "./components/UseEffect/StoreApi";

import Prueba from "./components/UseEffect/Prueba";
import { CounterApp } from "./components/Contador/CounterApp";

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    <CounterApp />
  </React.StrictMode>
);
