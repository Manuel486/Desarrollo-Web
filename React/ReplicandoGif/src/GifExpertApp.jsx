import { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';
import { Container } from 'react-bootstrap';

export const GifExpertApp = () => {
  
  const [categories, setcategories] = useState([]);
  
  const onAddCategory = (value) => {

    if(categories.includes(value)) return;

    setcategories([value,...categories]);
  }

  return (
    <Container>
      <h1 className='text-center'>Gif</h1>
      <AddCategory onAddCategory= { value => onAddCategory(value)} />
      {
        categories.map( category => (
          <GifGrid key={ category } category={ category } />
        ))
      }
    </Container>
  )
}
