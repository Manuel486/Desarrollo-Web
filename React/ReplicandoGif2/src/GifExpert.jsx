import { useState } from 'react'
import { GifGrid } from './components/GifGrid';
import { GifAdd } from './components/GifAdd';
import { Container } from '@mui/material';
import Swal from 'sweetalert2';

export const GifExpert = () => {

  const [categories, setCategories] = useState([]);
 
  const addCategory = (value) => {
    if( categories.includes(value) ){
      Swal.fire({
        icon: 'warning',
        title: 'Repetido',
        text: `La palabra ${value} ya esta incluida`,
      })
      
    }
    setCategories([value,...categories]);
  }

  return (
    <>
      <Container maxWidth="xl">
      <h1 style={{ textAlign: 'center' }}>GIF</h1>
        <GifAdd addCategory = { value => addCategory(value) } />
        {
          categories.map( (category) => (
            <GifGrid key={ category } category = { category } />
          ))
        }
      </Container>
    </>
  )
}
