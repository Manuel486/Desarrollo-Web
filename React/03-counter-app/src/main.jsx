import React from "react";
import ReactDOM, { createRoot } from "react-dom/client";
import { HelloWorldApp } from "./HelloWorldApp";
import { FirstApp } from "./FirstApp";

import "./styles.css";
import { CounterApp } from "./CounterApp";

// Forma deprecada
// const saludo = <h1>Hola mundo</h1>;
// const root = createRoot(document.getElementById('root'));
// root.render(saludo);

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    {/* <HelloWorldApp /> */}
    {/* <FirstApp title="Hola, Soy Vegeta" /> */}
    <CounterApp value={20} />
  </React.StrictMode>
);
