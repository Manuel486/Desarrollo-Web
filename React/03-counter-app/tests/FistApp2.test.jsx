import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en <FistApp />', () => { 

  const title = 'Hola, Soy Goku';
  const subtitle = 'Hola, Soy un subtitulo';

  test('debe de hacer un match con el snapshot', () => {

    const { container } = render(<FirstApp title={ title }/>)
    expect( container ).toMatchSnapshot();
  });


  test('debe de mostrar el mensaje "Hola, Soy Goku', () => {

    //screen.debug(); //almacena el body , este va renderizando cada cosa que renderizo
    render(<FirstApp title={ title }/>)
    expect( screen.getByText(title) ).toBeTruthy();
    // expect( screen.getByText(title) ).not.toBeTruthy();
  })

  test('debe de mostrar el titulo en un h1', () => { 
    render(<FirstApp title={ title }/>);
    expect( screen.getByRole('heading', { level: 1 }).innerHTML).toContain( title );
   })


  test('debe de mostrar el subtitulo enviado por props',()=>{
    render(<FirstApp title={ title } subTitle={subtitle}/>)
    //screen.debug();
    expect(screen.getAllByText(subtitle).length).toBe(2);
  })

 });