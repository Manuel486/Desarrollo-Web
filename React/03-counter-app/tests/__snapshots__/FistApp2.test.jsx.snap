// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FistApp /> debe de hacer un match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, Soy Goku
  </h1>
  <p />
  <p />
  <p>
    Manuel Chunga
  </p>
</div>
`;
